### ISSUE TITLE
[vcpkg] Building with g++-6 fails due to linker errors

### ISSUE THREAD

**Issue by strega-nil** [microsoft/vcpkg#13283]
```
libvcpkglib.a(sourceparagraph.cpp.o): In function `vcpkg::serialize_manifest_impl(vcpkg::SourceControlFile const&, bool)::{lambda(vcpkg::Json::Array&, vcpkg::Dependency const&)#4}::operator()(vcpkg::Json::Array&, vcpkg::Dependency const&) const':
sourceparagraph.cpp:(.text+0x3bf9): undefined reference to `vcpkg::DependencyField::FEATURES'
sourceparagraph.cpp:(.text+0x3c00): undefined reference to `vcpkg::DependencyField::FEATURES'
sourceparagraph.cpp:(.text+0x3c2d): undefined reference to `vcpkg::DependencyField::PLATFORM'
sourceparagraph.cpp:(.text+0x3c37): undefined reference to `vcpkg::DependencyField::PLATFORM'
libvcpkglib.a(sourceparagraph.cpp.o): In function `vcpkg::serialize_manifest_impl(vcpkg::SourceControlFile const&, bool)':
sourceparagraph.cpp:(.text+0x4454): undefined reference to `vcpkg::FeatureField::DESCRIPTION'
sourceparagraph.cpp:(.text+0x4461): undefined reference to `vcpkg::FeatureField::DESCRIPTION'
libvcpkglib.a(sourceparagraph.cpp.o): In function `vcpkg::LicenseExpressionField::visit_string[abi:cxx11](vcpkg::Json::Reader&, vcpkg::StringView, vcpkg::StringView)::{lambda()#1}::operator()() const':
sourceparagraph.cpp:(.text._ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv[_ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv]+0xd9): undefined reference to `vcpkg::LicenseExpressionField::EXPRESSION_WORDS'
sourceparagraph.cpp:(.text._ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv[_ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv]+0xe5): undefined reference to `vcpkg::LicenseExpressionField::EXPRESSION_WORDS'
sourceparagraph.cpp:(.text._ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv[_ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv]+0xfc): undefined reference to `vcpkg::LicenseExpressionField::VALID_LICENSES'
sourceparagraph.cpp:(.text._ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv[_ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv]+0x107): undefined reference to `vcpkg::LicenseExpressionField::VALID_LICENSES'
sourceparagraph.cpp:(.text._ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv[_ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv]+0x139): undefined reference to `vcpkg::LicenseExpressionField::VALID_EXCEPTIONS'
sourceparagraph.cpp:(.text._ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv[_ZZN5vcpkg22LicenseExpressionField12visit_stringB5cxx11ERNS_4Json6ReaderENS_10StringViewES4_ENKUlvE_clEv]+0x140): undefined reference to `vcpkg::LicenseExpressionField::VALID_EXCEPTIONS'
libvcpkglib.a(vcpkgcmdarguments.cpp.o): In function `vcpkg::VcpkgCmdArguments::imbue_from_environment()':
vcpkgcmdarguments.cpp:(.text+0x4250): undefined reference to `vcpkg::VcpkgCmdArguments::OVERLAY_PORTS_ENV'
vcpkgcmdarguments.cpp:(.text+0x4257): undefined reference to `vcpkg::VcpkgCmdArguments::OVERLAY_PORTS_ENV'
vcpkgcmdarguments.cpp:(.text+0x4354): undefined reference to `vcpkg::VcpkgCmdArguments::FEATURE_FLAGS_ENV'
vcpkgcmdarguments.cpp:(.text+0x435b): undefined reference to `vcpkg::VcpkgCmdArguments::FEATURE_FLAGS_ENV'
vcpkgcmdarguments.cpp:(.text+0x4401): undefined reference to `vcpkg::VcpkgCmdArguments::DEFAULT_VISUAL_STUDIO_PATH_ENV'
vcpkgcmdarguments.cpp:(.text+0x4408): undefined reference to `vcpkg::VcpkgCmdArguments::DEFAULT_VISUAL_STUDIO_PATH_ENV'
vcpkgcmdarguments.cpp:(.text+0x4563): undefined reference to `vcpkg::VcpkgCmdArguments::DISABLE_METRICS_ENV'
vcpkgcmdarguments.cpp:(.text+0x456a): undefined reference to `vcpkg::VcpkgCmdArguments::DISABLE_METRICS_ENV'
vcpkgcmdarguments.cpp:(.text+0x45cb): undefined reference to `vcpkg::VcpkgCmdArguments::TRIPLET_ENV'
vcpkgcmdarguments.cpp:(.text+0x45d2): undefined reference to `vcpkg::VcpkgCmdArguments::TRIPLET_ENV'
vcpkgcmdarguments.cpp:(.text+0x4633): undefined reference to `vcpkg::VcpkgCmdArguments::DOWNLOADS_ROOT_DIR_ENV'
vcpkgcmdarguments.cpp:(.text+0x463a): undefined reference to `vcpkg::VcpkgCmdArguments::DOWNLOADS_ROOT_DIR_ENV'
vcpkgcmdarguments.cpp:(.text+0x47b1): undefined reference to `vcpkg::VcpkgCmdArguments::VCPKG_ROOT_DIR_ENV'
vcpkgcmdarguments.cpp:(.text+0x47b8): undefined reference to `vcpkg::VcpkgCmdArguments::VCPKG_ROOT_DIR_ENV'
libvcpkglib.a(commands.info.cpp.o): In function `vcpkg::Commands::Info::InfoCommand::perform_and_exit(vcpkg::VcpkgCmdArguments const&, vcpkg::VcpkgPaths const&) const':
commands.info.cpp:(.text+0x67): undefined reference to `vcpkg::VcpkgCmdArguments::JSON_SWITCH'
libvcpkglib.a(metrics.cpp.o): In function `vcpkg::Metrics::generate_random_UUID()':
metrics.cpp:(.text+0x83c): undefined reference to `vcpkg::Metrics::append_hexits::hex'
metrics.cpp:(.text+0x895): undefined reference to `vcpkg::Metrics::append_hexits::hex'
metrics.cpp:(.text+0xa2c): undefined reference to `vcpkg::Metrics::append_hexits::hex'
metrics.cpp:(.text+0xa8c): undefined reference to `vcpkg::Metrics::append_hexits::hex'
libvcpkglib.a(vcpkgpaths.cpp.o): In function `vcpkg::VcpkgPaths::VcpkgPaths(vcpkg::Files::Filesystem&, vcpkg::VcpkgCmdArguments const&)':
vcpkgpaths.cpp:(.text+0x768a): undefined reference to `vcpkg::VcpkgCmdArguments::MANIFEST_MODE_FEATURE'
vcpkgpaths.cpp:(.text+0x7691): undefined reference to `vcpkg::VcpkgCmdArguments::MANIFEST_ROOT_DIR_ARG'
vcpkgpaths.cpp:(.text+0x7698): undefined reference to `vcpkg::VcpkgCmdArguments::FEATURE_FLAGS_ARG'
vcpkgpaths.cpp:(.text+0x769f): undefined reference to `vcpkg::VcpkgCmdArguments::FEATURE_FLAGS_ENV'
libvcpkglib.a(build.cpp.o): In function `vcpkg::Build::EnvCache::get_triplet_info[abi:cxx11](vcpkg::VcpkgPaths const&, vcpkg::Build::AbiInfo const&)':
build.cpp:(.text+0x4638): undefined reference to `vcpkg::VcpkgCmdArguments::COMPILER_TRACKING_FEATURE'
build.cpp:(.text+0x4646): undefined reference to `vcpkg::VcpkgCmdArguments::FEATURE_FLAGS_ARG'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Comment by strega-nil** [comment]
cc @BillyONeal we need to start building with our supported compilers in CI

Also, I'll fix this once #13038 is merged

**Comment by strega-nil** [comment]
Fixed by #13290
